### ISSUE TITLE
failure: nixos.tests.boot.biosUsb: keeps booting after USB transfer failure

### ISSUE THREAD

**Issue by raboof** [NixOS/nixpkgs#172922]
https://hydra.nixos.org/build/176455602/nixlog/3

Running `nix-build '<nixpkgs/nixos/tests/boot.nix>' -A biosUsb` on nixpkgs e680b5 doesn't show the problem on my machine, unfortunately, so it seems some kind of indeterminism

```
machine # <<< NixOS Stage 1 >>>
machine # 
machine # /init: source: line 125: can't open '/nix/store/zb52zilvazsljqr1p20lyvzwv5bdicf2-mounts.sh': No such file or directory
machine # 
machine # An error occurred in stage 1 of the boot process, which must mount the
machine # root filesystem on `/mnt-root' and then start stage 2.  Press one
machine # of the following keys:
machine # 
machine #   r) to reboot immediately
machine #   *) to ignore the error and continue
```

line 125 of `nixpkgs/nixos/modules/system/boot/stage-1-init.sh` is `source @earlyMountScript@`

**Comment by raboof** [comment]
Ah, this is a result of the earlier error:

```
machine # Loading /boot/initrd...cpage out of range (5)
machine # processing error - resetting ehci HC
machine # CHS: Error 0c00 reading sector 86723 (43/0/36)
machine # EDD: Error 0c00 reading sector 86723
machine # CHS: Error 0c00 reading sector 96330 (47/25/4)
machine # EDD: Error 0c00 reading sector 96330
machine # ok
```

which suggests this is a duplicate of #170803 . Indeed this build was on e680b5 which does not qet have the fix from 218a48c8491968735af9bb2cd87475206a6310e1, so that checks out.

I'm surprised it tries to boot even though the USB transfer failed, though - is that impossible to detect?

**Comment by raboof** [comment]
> I'm surprised it tries to boot even though the USB transfer failed, though - is that impossible to detect?

hmm, or perhaps it recovered from that failure, but there was another undetected corruption caused by the same bug. let's close this.
